core/src/main/java/hudson/PluginStrategy.java

```diff
@@ -92,5 +92,5 @@ PluginWrapper createPluginWrapper(File archive)
      * @param dependee newly loaded plugin.
      * @since 1.557
      */
-    void updateDependency(PluginWrapper depender, PluginWrapper dependee);
+    default void updateDependency(PluginWrapper depender, PluginWrapper dependee) {}
```

Comment:
Logging should be retained, because we expect the implementation to do something 🐜 . It would be also great to have a kind of `@ShouldImplement` annotation with auto-checkers during the build